### PR TITLE
Add Python 3.12 to documentation

### PR DESCRIPTION
**Pull request by tancnle**
Note Python 3.12 support in documentation.

- https://beam.apache.org/get-started/quickstart-py/
- https://beam.apache.org/documentation/programming-guide/

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: website/www/site/content/en/get-started/quickstart-py.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ If you're interested in contributing to the Apache Beam Python codebase, see the
 
 {{< toc >}}
 
-The Python SDK supports Python 3.8, 3.9, 3.10 and 3.11. Beam 2.48.0 was the last release with support for Python 3.7.
+The Python SDK supports Python 3.8, 3.9, 3.10, 3.11 and 3.12. Beam 2.48.0 was the last release with support for Python 3.7.
```
Comment (tancnle) [inline website/www/site/content/en/get-started/quickstart-py.md:26]: Should we also drop 3.8 from this list? According to the `tox.ini` [file](https://github.com/apache/beam/blob/master/sdks/python/tox.ini#L20), 3.8 is no longer tested.

File: website/www/site/content/en/documentation/programming-guide.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ programming guide, take a look at the
 {{< language-switcher java py go typescript yaml >}}
 
 {{< paragraph class="language-py" >}}
-The Python SDK supports Python 3.8, 3.9, 3.10, and 3.11.
+The Python SDK supports Python 3.8, 3.9, 3.10, 3.11, and 3.12.
```
Comment (tancnle) [inline website/www/site/content/en/documentation/programming-guide.md:38]: Should we also drop 3.8 from this list? According to the `tox.ini` [file](https://github.com/apache/beam/blob/master/sdks/python/tox.ini#L20), 3.8 is no longer tested.

File: website/www/site/content/en/get-started/quickstart-py.md (hunk 1 above)
Comment (Abacn) [reply website/www/site/content/en/get-started/quickstart-py.md:26]: Fine for now and we should do this after Beam 2.61.0 release is out. Beam releases every 6 weeks while the website/ is deployed on daily bases cc: @jrmccluskey

**Review by Abacn — APPROVED** [review]
thank you!

**Pull request merged**